### ISSUE TITLE
VerifyError when extending class has var parameter

### ISSUE THREAD

**Issue by TjarkoG** [lampepfl/dotty#14422]
Possibly a dupplicat of https://github.com/lampepfl/dotty/issues/14265 or caused by the same Bug

## Compiler version

Compiler Version: scala 3.1.0 with oracle java 17.0.2 2022-01-18 LTS

## Minimized code

```scala
class A {
  class B(z: Any) {}
  class C(var y: Any) extends B(y) {}
  val c = new C("bar")
}

@main def test() =
  new A
```

## Output

```
Exception in thread "main" java.lang.VerifyError: Bad type on operand stack
Exception Details:
  Location:
    A$C.<init>(LA;Ljava/lang/Object;)V @25: invokevirtual
  Reason:
    Type uninitializedThis (current frame, stack[2]) is not assignable to 'A$C'
  Current Frame:
    bci: @25
    flags: { flagThisUninit }
    locals: { uninitializedThis, 'A', 'java/lang/Object' }
    stack: { uninitializedThis, 'A', uninitializedThis }
  Bytecode:
    0000000: 2a2c b500 112b c700 0bbb 0013 59b7 0016
    0000010: bf2a 2bb5 0018 2a2b 2ab6 001b b700 1db1
    0000020:                                        
  Stackmap Table:
    same_frame(@17)

	at A.<init>(Main.scala:4)
	at Main$package$.test(Main.scala:8)
	at test.main(Main.scala:7)
```

## Expectation

I would expect this to run like in scala 2.13

**Comment by smarter** [comment]
Yes that looks like a duplicate but it's good to know it happens with non-enums too.

**Comment by smarter** [comment]
actually this is already fixed in 3.1.1.

**Comment by som-snytt** [comment]
https://github.com/lampepfl/dotty/issues/13630
